### PR TITLE
Backend shared utils

### PR DESCRIPTION
**Pull request by maciejhirsz**
Extracted out from `backend/core`, to be used in `backend/shard` in future PRs.

Requires #337 to be merged to master first (this PR will be changed to target `master` instead of `mh-backend-core` when that happens).

### CODE REVIEW

File: backend/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 [workspace]
 members = [
     "core",
+    "shared",
```
Comment (dvdplm) [inline backend/Cargo.toml:4]: "common" is often used for these?

File: backend/core/src/shard/connector.rs [hunk 2]
```diff
@@ -30,8 +26,8 @@ pub struct ShardConnector {
     chain: Option<Addr<Chain>>,
     /// Mapping `ShardConnId` to `NodeId`
     nodes: DenseMap<NodeId>,
-    /// Buffer for constructing continuation messages
-    contbuf: BytesMut,
+    /// Container for handling continuation messages
```
Comment (dvdplm) [inline backend/core/src/shard/connector.rs:29]: "Wrapper"?

File: backend/shared/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+pub mod types;
```
Comment (dvdplm) [inline backend/shared/src/lib.rs:1]: License headers

**Review by dvdplm — APPROVED** [review]
Seems uncontroversial.

File: backend/core/src/node/connector.rs [hunk 4]
```diff
@@ -224,34 +200,18 @@ impl StreamHandler<Result<ws::Message, ws::ProtocolError>> for NodeConnector {
     fn handle(&mut self, msg: Result<ws::Message, ws::ProtocolError>, ctx: &mut Self::Context) {
         self.hb = Instant::now();
 
-        let data = match msg {
-            Ok(ws::Message::Ping(msg)) => {
+        let data = match msg.map(|msg| self.multipart.handle(msg)) {
```
Comment (TarikGul) [inline backend/core/src/node/connector.rs:203]: Very neat optimization.

**Review by TarikGul — APPROVED** [review]
LGTM, took me a bit to piece part 1, and 2 together, but looks great!

File: backend/core/src/feed.rs [hunk 5]
```diff
@@ -13,10 +13,23 @@ pub mod connector;
 
 use connector::Serialized;
 
-pub trait FeedMessage: Serialize {
+pub trait FeedMessage {
     const ACTION: u8;
 }
 
+pub trait FeedMessageWrite: FeedMessage {
```
Comment (dvdplm) [inline backend/core/src/feed.rs:20]: Missing docs (which seems to be a pretty common thing in the project in general).

File: backend/core/src/shard/connector.rs [hunk 6]
```diff
@@ -77,65 +73,31 @@ impl ShardConnector {
...
-            self.contbuf.extend_from_slice(&bytes);
-        } else {
-            log::error!("Continuation buffer overflow");
-            self.contbuf = BytesMut::new();
-        }
-    }
-
-    fn finish_frame(&mut self) -> Bytes {
-        mem::replace(&mut self.contbuf, BytesMut::new()).freeze()
+        // TODO: get `NodeId` for `ShardConnId` and proxy payload to `self.chain`.
```
Comment (niklasad1) [inline backend/core/src/shard/connector.rs:81]: dependent on another PR or can it be fixed in this one?

**Review by niklasad1 — COMMENTED** [review]
nice, clean

File: backend/core/src/shard/connector.rs (hunk 6 above)
Comment (maciejhirsz) [reply backend/core/src/shard/connector.rs:81]: This will land in the next PR :)

**Comment by maciejhirsz** [comment]
Killing this PR, had to rebase and squash to fix all the conflicts with reverts and changes post revert happening on master, made a new branch on `backend-shard-rebased` that fixes this and will open a new PR against master from there.

**Pull request closed without merging**